core/src/test/scala/bekindrewind/VcrClientSpec.scala

```diff
@@ -0,0 +1,63 @@
+package bekindrewind
+
+import munit._
+import io.circe.parser._
+import io.circe.syntax._
+
+import java.net.URI
+import java.nio.file._
+import java.time.OffsetDateTime
+import scala.jdk.CollectionConverters._
+
+class VcrClientSpec extends FunSuite {
+
+  test("Request and response are saved as JSON") {
+    val recordingPath = Files.createTempFile("test", ".json")
+    val client        = MockClient(recordingPath, RecordOptions.default, VcrMatcher(_ => true))
+    assert(client.previouslyRecorded.isEmpty)
+    assert(client.newlyRecorded.get().isEmpty)
+
+    val record = VcrRecord(
+      VcrRecordRequest("GET", new URI("https://example.com/foo.json"), "{}", Map.empty),
+      VcrRecordResponse(200, "ok", Map.empty, "{}"),
+      OffsetDateTime.parse("2100-05-06T12:34:56.789Z")
+    )
+    client.newlyRecorded.set(Vector(record))
+    client.save()
+    assert(client.previouslyRecorded.isEmpty)
+    assert(client.newlyRecorded.get().sizeIs == 1)
+
+    val savedJson = Files.readAllLines(recordingPath).asScala.mkString("")
```

Comment:
```suggestion
    val savedJson = Files.readString(recordingPath)
```

Reply:
Fixed in https://github.com/reibitto/be-kind-rewind/pull/7/commits/3a9fb53eee97307547dcf2742ec5323f3e42be30